### PR TITLE
Log signup attempts to a csv file

### PR DESCRIPTION
**Pull request by fhartwig**

### CODE REVIEW

**Comment by fisx** [comment]
travis says:

```
setup-Simple-Cabal-1.22.4.0-x86_64-linux-ghc-7.10.2: can't find source for
Data/Csv/Missing in src, dist/dist-sandbox-d106cceb/build/autogen
```

git add?  add to cabal file?  both?

File: thentos-core/src/Thentos.hs [hunk 1]
```diff
@@ -73,7 +73,11 @@ makeMain commandSwitch =
     let actionState = ActionState config rng connPool
         logPath     = config >>. (Proxy :: Proxy '["log", "path"])
         logLevel    = config >>. (Proxy :: Proxy '["log", "level"])
+        signupLogPath = config >>. (Proxy :: Proxy '["signup_log"])
     configLogger logPath logLevel
+    case signupLogPath of
+        Just path -> configSignupLogger path
+        Nothing -> return ()
```
Comment (fisx) [inline thentos-core/src/Thentos.hs]: What about making `configSignupLogger` take a `Maybe`?

File: thentos-core/src/Thentos/Action/Unsafe.hs [hunk 2]
```diff
@@ -19,12 +19,13 @@ import "cryptonite" Crypto.Random (ChaChaDRG, DRG(randomBytesGenerate))
...
 import Data.Pool (withResource)
 import Data.Proxy (Proxy(Proxy))
-import Data.String.Conversions (LT, ST, SBS)
+import Data.String.Conversions (cs, LT, ST, SBS)
 import LIO.Core (liftLIO, getClearance, setClearanceP)
 import LIO.DCLabel (ToCNF, DCLabel, (%%), cFalse)
 import LIO.Label (lub)
 import LIO.TCB (Priv(PrivTCB), ioTCB)
-import System.Log (Priority(ERROR, CRITICAL))
+import System.Log (Priority(DEBUG, ERROR, CRITICAL))
```
Comment (fisx) [inline thentos-core/src/Thentos/Action/Unsafe.hs]: I think you're not using `DEBUG`.

File: thentos-core/src/Thentos/Action/Unsafe.hs [hunk 3]
```diff
@@ -142,3 +145,10 @@ renderTextTemplate :: ST -> MuContext IO -> UnsafeAction e s LT
 renderTextTemplate template context = liftIO $ hastacheStr hastacheCfg template context
   where
     hastacheCfg = defaultConfig { muEscapeFunc = emptyEscape }
+
+logSignupAttempt :: UserName -> UserEmail -> Bool -> UnsafeAction e s ()
+logSignupAttempt name email captchaCorrect = do
+    now <- getCurrentTime
+    let signupAttempt = SignupAttempt name email captchaCorrect now
+        logLine = cs . Builder.toByteString $ CsvBuilder.encodeRecord signupAttempt
+    liftIO $ logM signupLogger CRITICAL (init logLine)
```
Comment (fisx) [inline thentos-core/src/Thentos/Action/Unsafe.hs]: I'd prefer `INFO`.

File: thentos-core/src/Thentos/Types.hs [hunk 4]
```diff
@@ -96,6 +97,8 @@ import Database.PostgreSQL.Simple.TypeInfo (typoid)
 import Data.ByteString.Builder (doubleDec)
 import Data.ByteString.Conversion (ToByteString)
 import Data.Char (isAlpha)
+import Data.Csv ((.:))
+import Data.Csv.Missing ()
```
Comment (fisx) [inline thentos-core/src/Thentos/Types.hs]: This module is missing.  (-:

`git add`, I think.

**Comment by fisx** [comment]
looks good otherwise.  once the missing csv file is there, can you reference the resp. pr over at cassava in a comment?

**Comment by fhartwig** [comment]
I addressed all these points now, I think.

**Comment by fisx** [comment]
I just realized there is not test.  I think I'll add one before merging.

**Comment by fisx** [comment]
Thanks!  I made two minor changes.  If you have no objections, I will merge if travis is done (expecting purescript to break there, will ignore).

**Comment by fhartwig** [comment]
:+1:

**Comment by fisx** [comment]
(one of the changes broke things.  rebased!  merging!)

**Pull request merged**